### PR TITLE
import address from common

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR fixes import circularities:
- moves address type to common.py
- move all todos into issues or remove where they have been addressed/won't be addressed

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

**Review by solarw — APPROVED** [review]
LGTM

File: aea/aea.py [hunk 1]
```diff
@@ -229,7 +229,6 @@ def _get_msg_and_handlers_for_envelope(
     ) -> Tuple[Optional[Message], List[Handler]]:
         protocol = self.resources.get_protocol(envelope.protocol_id)
 
-        # TODO specify error handler in config and make this work for different skill/protocol versions.
```
Comment (DavidMinarsch) [inline aea/aea.py:232]: there is an open issue for this

File: aea/cli/run.py [hunk 2]
```diff
@@ -132,6 +132,4 @@ def _build_aea(
     except AEAPackageLoadingError as e:  # pragma: nocover
         raise click.ClickException("Package loading error: {}".format(str(e)))
     except Exception as e:
-        # TODO use an ad-hoc exception class for predictable errors
```
Comment (DavidMinarsch) [inline aea/cli/run.py:135]: there is an open issue for that

File: aea/cli_gui/__init__.py [hunk 3]
```diff
@@ -107,7 +107,7 @@ def get_agents() -> List[Dict]:
                     "public_id": tail,  # it is not a public_id actually, just a folder name.
                     # the reason it's called here so is the view that is used to represent items with public_ids
                     # used also for agent displaying
-                    # TODO: change it when we will have a separate view for an agent.
+                    # change it when we will have a separate view for an agent.
```
Comment (DavidMinarsch) [inline aea/cli_gui/__init__.py:110]: no need to keep a todo; if we ever do it the comment is there.

File: aea/configurations/loader.py [hunk 4]
```diff
@@ -209,7 +209,6 @@ def _validate(self, json_data: Dict) -> None:
         self.validate(json_data)
 
         expected_type = self.configuration_class.package_type
-        # TODO 'type' is optional for backward compatibility
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:212]: opened an issue for this and other configuration todos

File: aea/crypto/cosmos.py [hunk 5]
```diff
@@ -67,7 +67,6 @@ def is_transaction_settled(tx_receipt: Any) -> bool:
         """
         is_successful = False
         if tx_receipt is not None:
-            # TODO: quick fix only, not sure this is reliable
             is_successful = True
```
Comment (DavidMinarsch) [inline aea/crypto/cosmos.py:70]: nothing to do here

File: aea/crypto/ethereum.py [hunk 6]
```diff
@@ -481,7 +481,7 @@ def get_deploy_transaction(  # pylint: disable=arguments-differ
             "from": deployer_address,  # only 'from' address, don't insert 'to' address!
             "value": value,  # transfer as part of deployment
             "gas": gas,
-            "gasPrice": self.api.eth.gasPrice,  # TODO: refine
+            "gasPrice": self.api.eth.gasPrice,
```
Comment (DavidMinarsch) [inline aea/crypto/ethereum.py:484]: nothing to do here

File: aea/crypto/wallet.py [hunk 7]
```diff
@@ -91,8 +91,6 @@ class Wallet:
 
     """
 
-    # TODO do some check after loading the keys
```
Comment (DavidMinarsch) [inline aea/crypto/wallet.py:94]: if we ever want to do extra checks we can add a new issue; atm this is not a priority

File: aea/helpers/search/models.py [hunk 8]
```diff
@@ -610,7 +610,7 @@ def check_validity(self):
         :return ``None``
         :raises ValueError: if the object does not satisfy some requirements.
         """
-        if len(self.constraints) < 2:  # pragma: nocover  # TODO: do we need this check?
+        if len(self.constraints) < 2:  # pragma: nocover
```
Comment (DavidMinarsch) [inline aea/helpers/search/models.py:613]: we ported this from oef sdk, I don't see a reason to change it atm

File: aea/multiplexer.py [hunk 9]
```diff
@@ -40,10 +40,6 @@
 from aea.protocols.base import Message
 
 
-# TODO refactoring: this should be an enum
```
Comment (DavidMinarsch) [inline aea/multiplexer.py:43]: Yuri is working on it

File: aea/protocols/base.py [hunk 10]
```diff
@@ -92,7 +92,7 @@ def sender(self) -> Address:
     @sender.setter
     def sender(self, sender: Address) -> None:
         """Set the sender of the message."""
-        # enforce(self._sender is None, "Sender already set.")  # TODO: check if we can enable # noqa: E800
+        enforce(self._sender is None, "Sender already set.")
```
Comment (DavidMinarsch) [inline aea/protocols/base.py:95]: Messages should be immutable; let's see if the tests will pass

File: aea/skills/base.py [hunk 11]
```diff
@@ -280,7 +280,6 @@ def configuration(self) -> SkillComponentConfiguration:
             raise ValueError("Configuration not set.")  # pragma: nocover
         return self._configuration
 
-    # TODO consider rename this property
```
Comment (DavidMinarsch) [inline aea/skills/base.py:283]: again. not worthy of a todo

File: aea/skills/tasks.py [hunk 12]
```diff
@@ -115,7 +115,6 @@ def init_worker() -> None:
     :return: None
     """
     signal.signal(signal.SIGINT, signal.SIG_IGN)
-    # TODO check how to disable CTRL_C_EVENT signal on Windows: signal.signal(signal.CTRL_C_EVENT, signal.SIG_IGN)
```
Comment (DavidMinarsch) [inline aea/skills/tasks.py:118]: Yuri probably solved it, otherwise we need an issue

File: aea/test_tools/test_cases.py [hunk 13]
```diff
@@ -854,7 +854,6 @@ def setup_class(cls):
         """Set up the test class."""
         # make paths absolute
         cls.path_to_aea = cls.path_to_aea.absolute()
-        # TODO: decide whether to keep optionally: cls.packages_dir_path = cls.packages_dir_path.absolute()
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:857]: removed; either we do or we don't. Discussions in code is not a good idea.

File: examples/gym_ex/proxy/env.py [hunk 14]
```diff
@@ -136,10 +137,7 @@ def render(self, mode="human") -> None:
 
         :return: None
         """
-        # TODO: adapt this line to the new APIs. We no longer have a mailbox.
-        self._agent.mailbox._connection.channel.gym_env.render(  # pylint: disable=protected-access,no-member
-            mode
-        )
+        self._agent.runtime.multiplexer.default_connection.channel.gym_env.render(mode)
```
Comment (DavidMinarsch) [inline examples/gym_ex/proxy/env.py:140]: This should be the new api imo

File: packages/fetchai/connections/p2p_libp2p_client/connection.py [hunk 15]
```diff
@@ -103,7 +103,6 @@ def __init__(self, **kwargs):
                     ledger_id, SUPPORTED_LEDGER_IDS
                 )
             )
-        # TODO: ensure ledger_id matches with provided key of p2p node.
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p_client/connection.py:106]: done framework side

File: packages/fetchai/connections/tcp/tcp_client.py [hunk 16]
```diff
@@ -91,7 +91,7 @@ async def receive(self, *args, **kwargs) -> Optional["Envelope"]:
                 self.logger.debug("[{}] No data received.".format(self.address))
                 return None
             self.logger.debug("[{}] Message received: {!r}".format(self.address, data))
-            envelope = Envelope.decode(data)  # TODO handle decoding error
+            envelope = Envelope.decode(data)
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/tcp/tcp_client.py:94]: again; no need to keep reference here. obvious from code

File: packages/fetchai/skills/ml_data_provider/strategy.py [hunk 17]
```diff
@@ -78,7 +78,7 @@ def __init__(self, **kwargs) -> None:
 
         super().__init__(**kwargs)
         # loading ML dataset
-        # TODO this should be parametrized
+        # (this could be parametrized)
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/ml_data_provider/strategy.py:81]: not a todo; just an idea

File: packages/fetchai/skills/ml_train/ml_model.py [hunk 18]
```diff
@@ -40,7 +40,7 @@ def __init__(self, **kwargs):
         )
         super().__init__(**kwargs)
 
-        # TODO this at the moment does not work - need to compile the model according to the network configuration
+        # this at the moment does not work - need to compile the model according to the network configuration
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/ml_train/ml_model.py:43]: not a todo; comment is enough

File: packages/fetchai/skills/tac_negotiation/helpers.py [hunk 19]
```diff
@@ -138,7 +138,6 @@ def build_goods_query(
     constraint_expr = cast(List[ConstraintExpr], constraints)
 
     if len(good_ids) > 1:
-        # TODO: fix  # noqa: E800
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/tac_negotiation/helpers.py:141]: nothing to fix

**Pull request merged**